docs/remote-interface.md

```diff
@@ -0,0 +1,59 @@
+# Remote interface
+
+## Introduction
+
+```
+ +---------------------------------------------+
+ |                   Client                    |
+ |  +------------------+  +------------------+ |
+ |  |  send new work   |  |    get result    | |
```

Comment:
The essential operations should not contain only "send new work" and "get result". We need more descriptions.